### PR TITLE
Do not rely on conversions between float and extended floating point types

### PR DESCRIPTION
**Pull request by miscco**
The issue we have is that our tests rely extensively on those conversions which makes it incredibly painfull to test

Fixes nvbug4739601

### CODE REVIEW

File: libcudacxx/include/cuda/std/__cuda/cmath_nvbf16.h [hunk 1]
```diff
@@ -37,47 +37,47 @@ _LIBCUDACXX_BEGIN_NAMESPACE_STD
 // trigonometric functions
 inline _LIBCUDACXX_INLINE_VISIBILITY __nv_bfloat16 sin(__nv_bfloat16 __v)
 {
-  NV_IF_ELSE_TARGET(NV_IS_DEVICE, (return ::hsin(__v);), (return __nv_bfloat16(::sin(float(__v)));))
+  NV_IF_ELSE_TARGET(NV_IS_DEVICE, (return ::hsin(__v);), (return __float2bfloat16(::sin(__bfloat162float(__v)));))
```
Comment (bernhardmgruber) [inline libcudacxx/include/cuda/std/__cuda/cmath_nvbf16.h:40]: Important: The host path does not seem right. The code calls the C library's `::sin` here, which promotes the argumet to a `double`. We should either call `::sinf` or `std::sin` here.

Applies to several math functions below.

**Review by bernhardmgruber — COMMENTED** [review]
LGTM, except for:

**Review by griwes — APPROVED** [review]
LGTM. @bernhardmgruber is right about the host paths I think, but this was preexisting condition of how I wrote that code in the first place, so while we should fix that, it shouldn't block this.

File: libcudacxx/include/cuda/std/__cuda/cmath_nvbf16.h (hunk 1 above)
Comment (bernhardmgruber) [reply libcudacxx/include/cuda/std/__cuda/cmath_nvbf16.h:40]: Created issue: #2078

**Review by bernhardmgruber — APPROVED** [review]

**Pull request merged**